pkg/search/postgres/store.go

```diff
@@ -0,0 +1,116 @@
+package postgres
+
+import (
+	"context"
+	"time"
+
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	ops "github.com/stackrox/rox/pkg/metrics"
+	"github.com/stackrox/rox/pkg/postgres"
+	"github.com/stackrox/rox/pkg/postgres/walker"
+	"github.com/stackrox/rox/pkg/search"
+)
+
+// PermissionChecker is a permission checker that could be used by GenericStore
+type PermissionChecker interface {
+	ExistsAllowed(ctx context.Context) (bool, error)
+}
+
+type primaryKeyGetter[T any, PT unmarshaler[T]] func(obj PT) string
+type durationTimeSetter func(start time.Time, op ops.Op)
```

Comment:
I'm open to discussion on that one, as the change is not necessarily bringing a big gain.
```suggestion
type durationTimeSetter func(start time.Time, op ops.Op, t string)
```

This would allow, if the GenericStore also stores the trimmed type (additional field `typ string`) to move the implementation of `metricsSetAcquireDBConnDuration` and `metricsSetPostgresOperationDurationTime` to the generic file.
Since the `metrics` package where the time collection functions are defined is in central, the functions would still have to be passed to the generic store constructor, though.

Reply:
> Since the metrics package where the time collection functions are defined is in central, the functions would still have to be passed to the generic store constructor, though.

Exactly that's why I created this functions as the dependency form pkg to central is not allowed.

Reply:
I realized that we don't need start.Time and changed this in 
- https://github.com/stackrox/stackrox/pull/6778